homeassistant/components/script/__init__.py

```diff
@@ -335,6 +339,16 @@ async def async_turn_on(self, **kwargs):
         self.hass.async_create_task(coro)
         await self._changed.wait()
 
+    async def _async_run(self, variables, context):
+        with trace_script(
```

Comment:
Do we need to create a new context here?

Reply:
No. Script context is passed in, that's fine.